src/app/utilities/transmart-utilities/transmart-constraint-serialiser.ts

```diff
@@ -60,23 +60,24 @@ export class TransmartConstraintSerialiser extends AbstractConstraintVisitor<obj
   /**
    * Wrap a given query object with subselection clause
    * @param {Object} queryObject
+   * @param {string} dimension - dimension for the subselection constraint
    * @returns {Object}
    */
-  static wrapWithSubselection(queryObject: object): object {
+  static wrapWithSubselection(queryObject: object, dimension: string): object {
```

Comment:
This class may be further refactored. I don't see a need for `combination.mark` in `visitCombinationConstraint`. If the combination constraint has a subjectdimension set, then all its children are wrapped in a subselection.
This can be done by, e.g., 
```groovy
return {
  type: combination.combinationState === CombinationState.And ? 'and' : 'or',
  args: combination.args.map(arg => this.visit(new SubselectionConstraint(combination.dimension, args)))
```
or something like that.